src/witan/gias.clj

```diff
@@ -0,0 +1,731 @@
+           "LA (code)"                   #_:la-code                       :string
+           "EstablishmentNumber"         #_:establishment-number          :string
+           "OpenDate"                    #_:open-date                     [:packed-local-date parse-date]
+           "CloseDate"                   #_:close-date                    [:packed-local-date parse-date]
+           "CensusDate"                  #_:census-date                   [:packed-local-date parse-date]
+           "UKPRN"                       #_:ukprn                         :string
+           "FEHEIdentifier"              #_:fehe-identifier               :string
+           "OfstedLastInsp"              #_:ofsted-last-insp              [:packed-local-date parse-date]
+           "LastChangedDate"             #_:last-changed-date             [:packed-local-date parse-date]
+           "TelephoneNum"                #_:telephone-num                 :string
+           "DateOfLastInspectionVisit"   #_:date-of-last-inspection-visit [:packed-local-date parse-date]
+           "NextInspectionVisit"         #_:next-inspection-visit         [:packed-local-date parse-date]
+           "PreviousLA (code)"           #_:previous-la-code              :string
+           "PreviousEstablishmentNumber" #_:previous-establishment-number :string
+           "UPRN"                        #_:uprn                          :string
+           "QABReport"                   #_:qab-report                    :string
+           "CHNumber"                    #_:ch-number                     :string})))
+
+(def key-fn
+  "Default key-fn to be applied to edubasealldata.csv column names to obtain dataset column names."
+  #(csv-col-name->col-name % %))
+
+(def parser-fn
+  "parser-fn for ds/->dataset read of edubasealldata.csv with `key-fn` applied to CSV column names."
+  (into (sorted-map-by (partial compare-mapped-keys col-name->order))
+        (update-keys csv-parser-fn key-fn)))
+
+(defn ->ds
+  "Read GIAS all establishment data into a dataset.
+   Use optional `options` map to specify:
+   - CSV file to read: via ::file-path or ::resource-file-name (for files in resource folder).
+     [Defaults to ::resource-file-name `default-resource-file-name`.]
+   - Additional or over-riding options for `->dataset`."
+  ([] (->ds {}))
+  ([{::keys [resource-file-name file-path]
+     :or    {resource-file-name default-resource-file-name}
+     :as    options}]
+   (with-open [in (-> (or file-path (io/resource resource-file-name))
+                      io/file
+                      io/input-stream)]
+     (ds/->dataset in (merge {:file-type    :csv
+                              :separator    ","
+                              :dataset-name (or file-path resource-file-name)
+                              :header-row?  true
+                              :key-fn       key-fn
+                              :parser-fn    parser-fn}
```

Comment:
This would be where we'd translate allow/block list to the underlying tmd anachronism

Reply:
Translation added and examples updated to use `:column-allowlist`.